rust/ql/lib/codeql/rust/elements/internal/FormatArgumentImpl.qll

```diff
@@ -40,6 +40,16 @@ module Impl {
 
     override Format getParent() { result = Synth::TFormat(parent, index, _, _) }
 
+    /** Gets the position of this argument. */
```

Comment:
This predicate might be a bit confusing. I think it should be dropped.

There are named and positional arguments which can be mixed to some extent. Not sure what the "position" of a named argument would mean, and it would be strange if `getPosition` does not coincide with the position of a positional argument. For example:
```rust
let width = 2;
println!("Value {value:#precision$.0$}", width, precision = 3, value = 100);
```
would have `getPosition() = 2` for `0` .